portal/src/util/pagination.ts

```diff
@@ -0,0 +1,122 @@
+export interface GetPaginationRenderDataInput {
+  offset: number;
+  pageSize: number;
+  totalCount?: number;
+}
+
+export interface PaginationRenderData {
+  currentOffset: number;
+  offsets: number[];
+  firstPageButtonEnabled: boolean;
+  prevPageButtonEnabled: boolean;
+  nextPageButtonEnabled: boolean;
+  lastPageButtonEnabled: boolean;
+  maxOffset?: number;
+}
+
+const PAGE_OFFSET = 3;
```

Comment:
I think we should add an explanation on what it is.
If I understood correctly, it is the number of pages before/after the current page shown in the page list?

Reply:
Yes.